app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+        nonlocal cache_storage
+        if args in cache_storage:
```

Comment:
Using `args` as a key in `cache_storage` can lead to issues if `args` contains mutable types like lists. Consider converting `args` to a hashable type, such as a tuple of tuples, to ensure it can be used as a dictionary key.